sahi/model.py

```diff
@@ -663,3 +664,167 @@ def _create_object_prediction_list_from_original_predictions(
         object_prediction_list_per_image = [object_prediction_list]
 
         self._object_prediction_list_per_image = object_prediction_list_per_image
+
+
+class TorchVisionDetectionModel(DetectionModel):
+    def load_model(self):
+        try:
+            import torchvision
+        except ImportError:
+            raise ImportError(
+                "torchvision is not installed. Please run 'pip install -U torchvision to use this "
+                "torchvision models'"
+            )
+
+        # set model
+        try:
+            from sahi.utils.torch import torch
+
+            model = self.config_path
+            model.load_state_dict(torch.load(self.model_path))
+            model.eval()
+            model = model.to(self.device)
+            self.model = model
+        except Exception as e:
+            raise Exception(f"Failed to load model from {self.model_path}. {e}")
+
+        # set category_mapping
+        from sahi.utils.torchvision import COCO_CLASSES
+
+        if self.category_mapping is None:
+            category_names = {str(i): COCO_CLASSES[i] for i in range(len(COCO_CLASSES))}
+            self.category_mapping = category_names
+
+    def perform_inference(self, image: np.ndarray, image_size: int = None):
+        """
+        Prediction is performed using self.model and the prediction result is set to self._original_predictions.
+        Args:
+            image: np.ndarray
+                A numpy array that contains the image to be predicted. 3 channel image should be in RGB order.
+            image_size: int
+                Inference input size.
+        """
+        try:
+            import torchvision
+        except ImportError:
+            raise ImportError("torchvision is not installed. Please install torchvision to use this model.")
+
+        # Confirm model is loaded
+        assert self.model is not None, "Model is not loaded, load it by calling .load_model()"
+
+        from sahi.utils.torchvision import numpy_to_torch, resize_image
```

Comment:
Import statement'ı, file'ın üst kısmında olmalı

Reply:
@devrimcavusoglu DetectionModel implementasyonlarinin hard dependency olmasini istemedigimiz icin bu sekilde lazy load yapiyoruz. yolov5 ile detection yapacak kisi bu utili import etmese de olur. Ne dusunuyorsun bu durum icin?

Reply:
paket genelinde bakarsak bence bu şekilde top-level dependency olmasında zarar yok bence. Kullanmayacak user'a çok overhead ve storage açısından problem yaratmıyorsa (böyle olduğunu varsaymıştım) paket seviyesinde dependency olmasının dezavantajı olmaz bence.

Reply:
mmdet ile kullanmak isteyen birinin yolov5 kurmadan kullanabilmesini sagliyoruz su anda, o yuzden bu modelleri birbirinden soyutlamaya calisiyoruz, genelde kullanicilar sahi'yi tek bir turde model ile kullanmak istiyor

Reply:
Tüm `model.py`a bakınca daha iyi anladım şimdi, idealde bu detection modelleri belki factory class'tan generate edilecek şekilde ayrı modüllerde yazmak isteyebiliriz belki. İleride code refactoring için düşünülebilir potansiyel olarak.